kernel/pci/src/lib.rs

```diff
@@ -215,44 +212,52 @@ impl PciLocation {
     }
 
     /// read 32-bit data at the specified `offset` from the PCI device specified by the given `bus`, `slot`, `func` set.
-    pub fn pci_read_32(&self, offset: u16) -> u32 {
+    pub fn pci_read_32(&self, offset: u8) -> u32 {
         unsafe { 
             PCI_CONFIG_ADDRESS_PORT.lock().write(self.pci_address(offset)); 
         }
-        PCI_CONFIG_DATA_PORT.lock().read() >> ((offset & (!PCI_CONFIG_ADDRESS_OFFSET_MASK)) * 8)
+        Self::data_port_read() >> ((offset & (!PCI_CONFIG_ADDRESS_OFFSET_MASK)) * 8)
     }
 
     /// Read 16-bit data at the specified `offset` from this PCI device.
-    pub fn pci_read_16(&self, offset: u16) -> u16 {
+    pub fn pci_read_16(&self, offset: u8) -> u16 {
         self.pci_read_32(offset) as u16
     } 
 
     /// Read 8-bit data at the specified `offset` from the PCI device.
-    pub fn pci_read_8(&self, offset: u16) -> u8 {
+    pub fn pci_read_8(&self, offset: u8) -> u8 {
         self.pci_read_32(offset) as u8
     }
 
     /// Write 32-bit data to the specified `offset` for the PCI device.
-    pub fn pci_write(&self, offset: u16, value: u32) {
+    pub fn pci_write(&self, offset: u8, value: u32) {
         unsafe { 
             PCI_CONFIG_ADDRESS_PORT.lock().write(self.pci_address(offset)); 
-            PCI_CONFIG_DATA_PORT.lock().write((value) << ((offset & 2) * 8));
+            Self::data_port_write((value) << ((offset & 2) * 8));
         }
     }
 
+    fn data_port_write(value: u32) {
+        unsafe {
+            PCI_CONFIG_DATA_PORT.lock().write(value);
+        }
+    }
+
+    fn data_port_read() -> u32 {
+        PCI_CONFIG_DATA_PORT.lock().read()
+    }
```

Comment:
I don't think these are much of an improvement, they just hide what's going on. Probably best to delete since the existing code was minimal anyway.

Reply:
I can read it better without a lot of `unsafe {}` everywhere and it is similar to the ps2 code.
Oh and I should make this a little clearer: I have some more local changes which also adjust the address port reads/writes (then we have even less unsafe blocks)

Reply:
I'll accept it this time because they're private functions, but in general, hiding unsafe blocks without actually assuring the safety of an outside invocation is bad practice